test/client/integration.test.ts

```diff
@@ -45,6 +45,11 @@ describe("Client Integration", function() {
         const apiPacker = CoreAPI.new(storagePacker, genesis, provider, signer);
         const apiSyncer = CoreAPI.new(storageSyncer, genesis, provider, signer);
 
+        const groupToRegister = Group.new({ n: 1 });
+        await apiPacker.contracts.blsAccountRegistry.register(
+            groupToRegister.getUser(0).pubkey
+        );
```

Comment:
Is there a preferred abstraction/way of doing this test data setup, suck as the packer command below?

Reply:
Haven't developed an opinion on this. I now see some options here, but leave you to decide. It's also fine to do a quick and dirty one here as long as it works.

- Add a register method in Group. `await group.registerKeyOneByOne()`
  - pros: looks straight forward
  - cons: need to pass API to group somehow. Group knows too much.
- Add registerKeyOneByOne to API, which takes group as an argument. `await api.registerKeyOneByOne(group)`
  - pros: Group stays agnostic to API
  - cons: API should reflect the node's functionality. When we implement Create2Transfer the coordinator does need to help to register a public key for a user. But in the test here we are using this method to do test data setup, which is not a functionality of the node, so it might confuse people.
- Add a third class/function to do test data setup `await registerPublicKey(api, group)`
  - pros: no extra responsibilities for API or Group
  - cons: It looks like we are just extracting a function for these two lines of code, not improving readability a lot. 

Let me also expand on TransferPackingCommand too. The coordinator needs to pack Deposit, Transfer, MassMigration, or Create2Transfer batches when they win a slot to propose. But each type of batch depends on some objects to pack. To unify the packing interface, we created BatchPackingCommand that has the packAndSubmit method. So that in the packer service, packer can select one of the commands and fire the packAndSubmit method without worrying about which objects the commands depend on.

Reply:
Thanks for the packing command explanation, that makes more sense now.

I think if this data setup gets more complicated I will go with the third option but for now I will [KISS](https://en.wikipedia.org/wiki/KISS_principle)